plum/resolver.py

```diff
@@ -20,30 +20,31 @@ class AmbiguousLookupError(LookupError):
 
     def __init__(
         self,
-        fname: Optional[str],
+        f_name: Union[str, None],
         target: Union[Tuple[object, ...], Signature],
-        methods: "MethodList",
+        methods: MethodList,
     ):
-        """Create a new NotFoundLookupError.
+        """Create a new :class:`AmbiguousLookupError`.
 
         Args:
-            fname (Optional[str]): name (or qualified name) of the function
-                that could not be resolved.
-            target (Union[Tuple[object, ...], Signature]): target signature
+            f_name (str or :obj:`None`): Name (or qualified name) of the function that
+                could not be resolved.
+            target (Union[Tuple[object, ...], :class:`.Signature`]): Target signature
                 or arguments that could not be resolved.
-            methods (MethodList): list of ambiguous methods.
+            methods (:class:`.MethodList`): List of ambiguous methods.
         """
-        self.fname = fname if fname is not None else "<function>"
+        self.f_name = f_name if f_name is not None else "<function>"
         self.target = target
         self.methods = methods
 
     def __rich_console__(self, console, options):
-        yield Text(f"{self.fname}{self.target} is ambiguous.")
-        yield Text()
-        yield Text("Valid matches are:")
+        yield Text(
+            f"{self.f_name}{self.target} is ambiguous between the following methods:"
+        )
```

Comment:
The blank line here was deliberate, to mimic Julia (and IMO to give better readability)

```python
julia> f(1,1)
ERROR: MethodError: f(::Int64, ::Int64) is ambiguous.

Candidates:
  f(x::Int64, y)
    @ Main REPL[1]:1
  f(x, y::Int64)
    @ Main REPL[2]:1

Possible fix, define
  f(::Int64, ::Int64)

Stacktrace:
```

and in the PR this yield 
```python
AmbiguousLookupError: f(1, 1) is ambiguous.

Valid matches are:
f(x: int, y: Any)
    <function f at 0x1064d45e0> @ <ipython-input-2-b5a79a933176>:1
f(x: Any, y: int)
    <function f at 0x2896e5580> @ <ipython-input-3-e0ee52c2c087>:1
```

Reply:
@PhilipVinc Ahhh, that's clear! Let me change that back. Thanks. :)

Reply:
(if however you prefer the more condensed version, do go ahead!)

Reply:
@PhilipVinc Do you know if we can indent the list of shown methods, like Julia does?

Reply:
```python
from rich.padding import Padding
from rich.columns import Columns

    def __rich_console__(self, console, options):
        yield Text(f"{self.f_name}{self.target} is ambiguous.")
        yield Text()
        yield Text("Candidates:")
        yield Padding(Columns([m.repr_mismatch() for m in self.methods]), (0,1))
```

the 1 is the horizontal padding

Reply:
I think Julia uses 2 as padding, but then the tab looks too ugly inside. I think we could then replace the 4 spaces with just 2 spaces in methods.py

```
        res.append("\n    ")
        res.append_text(repr_pyfunction(self.implementation))
```
with
```
        res.append("\n  ")
        res.append_text(repr_pyfunction(self.implementation))
```

Reply:
UI design is a rabbit hole

Reply:
> UI design is a rabbit hole

It really is...

Thanks for looking this up!! `Padding` works perfectly. I opted for 4 spaces to conform with `repr_pyfunction`.